src/client.js

```diff
@@ -105,6 +123,25 @@ class Client extends events.EventEmitter {
     this._interval = setInterval(
       () => this.recycle(),
       this._recycleInterval);
+    
+    if (this.pulse) {
+      this._reclaiminterval = setTimeout( 
+        callreclaim = () => {
+          const pulsecredentials = pulseCredentials();
+          this.connectionString = pulsecredentials.connectionString;
+          this.namespace = pulsecredentials.namespace;
+          this.reclaimAt = pulsecredentials.reclaimAt;
+          this.recycle();
```

Comment:
Here, too, it'd be better to just call `recycle` in the interval, knowing that when the instance eventually tries to reconnect, `reclaimAt` will be in the past and it will fetch a new connection string.